testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCCustomRegionTestCase.java

```diff
@@ -0,0 +1,397 @@
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.TWO_DEPLOYMENTS;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.TWO_NODES;
+
+/**
+ * Test of clustered JPA 2nd level cache implemented by Infinispan using entity custom region.
+ * <p>
+ * In persistence.xml we add the following:
+ * <code>
+ * <property name="hibernate.cache.infinispan.entity.cfg" value="entity-replicated"/>
+ * </code>
+ * This has the effect of using the Infinispan cache "entity-replicated" as the template for entities's caches;
+ * </p>
+ * <p>
+ * In the standalone-ha.xml, we have an invalidation-cache named "entity" in the cache container "hibernate" of
+ * the Infinispan subsystem;
+ * By default, this cache would be used as the template for entities's caches;
+ * Here we are overriding this default creating a replicated-cache in the cache container "hibernate", and using it
+ * as template;
+ * </p>
+ * <p>
+ * We are using the entity {@link DummyEntityCustomRegion}, which also defines a custom region name to be used for
+ * its second level cache name:
+ * <code>
+ *
+ * @author Jan Martiska and Tommaso Borgato
+ * @Cache(usage = CacheConcurrencyStrategy.TRANSACTIONAL, region = DummyEntityCustomRegion.DUMMY_ENTITY_REGION_NAME)
+ * </code>
+ * </p>
+ * <p>
+ * The result is that, for the entity {@link DummyEntityCustomRegion}, a replicated-cache named
+ * "ClusteredJPA2LCCustomRegionTestCase.war#MainPU.DUMMY_ENTITY_REGION_NAME" is created;
+ * </p>
+ * <p>
+ * The goals of this test are:
+ * <ul>
+ * <li>verify that the cache created for the entity {@link DummyEntityCustomRegion} is actually a replicated-cache</li>
+ * <li>verify that the cache created for the entity {@link DummyEntityCustomRegion} is actually named accordingly with the annotation on the entity</li>
+ * </ul>
+ * </p>
+ */
+@RunWith(Arquillian.class)
+public class ClusteredJPA2LCCustomRegionTestCase {
+
+    private static final String MODULE_NAME = ClusteredJPA2LCCustomRegionTestCase.class.getSimpleName();
+
+    @ArquillianResource
+    protected ContainerController controller;
+
+    @ArquillianResource
+    protected Deployer deployer;
+
+    @Deployment(name = DEPLOYMENT_1, managed = false, testable = false)
+    @TargetsContainer(NODE_1)
+    public static Archive<?> createDeploymentForContainer1() {
+        return createDeployment();
+    }
+
+    @Deployment(name = DEPLOYMENT_2, managed = false, testable = false)
+    @TargetsContainer(NODE_2)
+    public static Archive<?> createDeploymentForContainer2() {
+        return createDeployment();
+    }
+
+    private static Archive<?> createDeployment() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, MODULE_NAME + ".war");
+        war.addPackage(ClusteredJPA2LCCustomRegionTestCase.class.getPackage());
+        war.addAsWebInfResource(ClusteredJPA2LCCustomRegionTestCase.class.getPackage(), "persistence.xml",
```

Comment:
addAsManifestResource makes more sense I think

Reply:
@simkam I don't get this one

Reply:
Looks okay to me, as persistence.xml can be in the WEB-INF/classes directory of a WAR file (WEB-INF/classes/META-INF/persistence.xml).